master/reference/cni-plugin/configuration.md

```diff
@@ -180,3 +200,70 @@ When using the CNI `host-local` IPAM plugin, a special value `usePodCidr` is all
 ```
 
 When making use of the `usePodCidr` option, the Calico CNI plugin requires read-only Kubernetes API access to the `Nodes` resource.
+
+##### IPAM Manipulation with Kubernetes Annotations
+
+###### Specifying IP Pools on a per-Pod basis
+
+In addition to specifying IP Pools in the CNI config as discussed above, Calico IPAM supports specifying IP Pools per-Pod using the following [Kubernetes annotations](https://kubernetes.io/docs/user-guide/annotations/).
+
+- `cni.projectcalico.org/ipv4pools`: A list of configured IPv4 Pools from which to choose an address for the Pod.
+
+   Example:
+ 
+```yaml
+annotations:
+      "cni.projectcalico.org/ipv4pools": "[\"192.168.0.0/16\"]"
+```
+
+- `cni.projectcalico.org/ipv6pools`: A list of configured IPv6 Pools from which to choose an address for the Pod.
+
+   Example:
+ 
+```yaml
+annotations:
+      "cni.projectcalico.org/ipv6pools": "[\"2001:db8::1/120\"]"
+```
+
+If provided, these IP Pools will override any IP Pools specified in the CNI config.
+
+
+
+  > **Note:**
+  >
+  > This requires the IP Pools to exist before `ipv4pools` or `ipv6pools` annotations are used.
+  > Requesting a subset of an IP Pool is not supported. IP Pools requested in the annotations must be the exact match to the IP Pools created.
```

Comment:
minor wording tweak - "must be the exact match to the IP Pools created" would be better as "must exactly match a configured IP Pool"